### PR TITLE
don't test ldap when form isn't valid

### PR DESCRIPTION
**Pull request by edewit**
fixes: #1776

### CODE REVIEW

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
Form validation working as expected @edewit - nice fix!

Can we use the `helperTextInvalid` prop on the `connectionUrl` field? The helper text renders in black for this field but for the other fields the invalid text shows up red. Using `helperTextInvalid` seems to fix it! WDYT @edewit ?

**Comment by edewit** [comment]
Yes @jenny-s51 seems that parts of this form have some custom helper text rendering, updated this PR to change it to PF

File: src/user-federation/kerberos/KerberosSettingsRequired.tsx [hunk 1]
```diff
@@ -125,6 +125,10 @@ export const KerberosSettingsRequired = ({
           }
           fieldId="kc-kerberos-realm"
           isRequired
+          validated={
+            form.errors.config?.kerberosRealm?.[0] ? "error" : "default"
+          }
+          helperTextInvalid={form.errors.config.kerberosRealm[0].message}
```
Comment (jenny-s51) [inline src/user-federation/kerberos/KerberosSettingsRequired.tsx:131]:
```suggestion
          helperTextInvalid={form.errors.config.kerberosRealm?.[0].message}
```

File: src/user-federation/kerberos/KerberosSettingsRequired.tsx [hunk 2]
```diff
@@ -187,6 +191,8 @@ export const KerberosSettingsRequired = ({
           }
           fieldId="kc-key-tab"
           isRequired
+          validated={form.errors.config?.keyTab?.[0] ? "error" : "default"}
+          helperTextInvalid={form.errors.config.keyTab[0].message}
```
Comment (jenny-s51) [inline src/user-federation/kerberos/KerberosSettingsRequired.tsx:195]:
```suggestion
          helperTextInvalid={form.errors.config.keyTab?.[0].message}
```

**Review by jenny-s51 — CHANGES_REQUESTED** [review]
Thanks for refactoring those @edewit ! LDAP form is looking good -- there are just a couple areas in the Kerberos form where we should be using dot notation to prevent undefined errors

**Comment by edewit** [comment]
Good catch @jenny-s51 don't know why I didn't see that...

**Review by jenny-s51 — APPROVED** [review]
LGTM! 👍

**Pull request merged**